### ISSUE TITLE
why we need eps decay when using adadelta optimizer?

### ISSUE THREAD

**Issue by WanderingSven** [espnet/espnet#2648]
Hello,everyone
        When I use espnet to train las, I find the code will decay eps of adadelta optimizer. why we shoud do it? Is it necessary? when I use pretrained asr to train other task, should I use smaller eps than 1e-08?

code:
if args.opt == "adadelta":
        if args.criterion == "acc" and mtl_mode != "ctc":
            trainer.extend(
                restore_snapshot(
                    model, args.outdir + "/model.acc.best", load_fn=torch_load
                ),
                trigger=CompareValueTrigger(
                    "validation/main/acc",
                    lambda best_value, current_value: best_value > current_value,
                ),
            )
            trainer.extend(
                adadelta_eps_decay(args.eps_decay),
                trigger=CompareValueTrigger(
                    "validation/main/acc",
                    lambda best_value, current_value: best_value > current_value,
                ),
            )
        elif args.criterion == "loss":
            trainer.extend(
                restore_snapshot(
                    model, args.outdir + "/model.loss.best", load_fn=torch_load
                ),
                trigger=CompareValueTrigger(
                    "validation/main/loss",
                    lambda best_value, current_value: best_value < current_value,
                ),
            )
            trainer.extend(
                adadelta_eps_decay(args.eps_decay),
                trigger=CompareValueTrigger(
                    "validation/main/loss",
                    lambda best_value, current_value: best_value < current_value,
                ),
            )

**Comment by sw005320** [comment]
We're using the epsilon decay described in the original attention-based ASR paper https://arxiv.org/pdf/1508.04395.pdf
It definitely improves performance, but there would be other methods.
Also, for the use of fine-tuning from pre-trained models, I think you could use the original one (1e-08).
We confirmed that it would work with such fine-tuning scenarios, but please also try your idea since using smaller ones make sense.

**Comment by WanderingSven** [comment]
Thanks for your reply @sw005320
